flexmeasures/data/models/reporting/tests/test_pandas_reporter.py

```diff
@@ -159,3 +159,54 @@ def test_reporter_empty(setup_dummy_data):
     )
 
     assert report[0]["data"].empty
+
+
+def test_pandas_reporter_unit_conversion(app, setup_dummy_data):
+    """
+    Check that the unit conversion feature can handle the following cases:
+        - kW -> kW
+        - kW -> MW
+        - kW -> MWh
+    """
+    s1, s2, s3, s4, report_sensor, daily_report_sensor = setup_dummy_data
+
+    reporter_config = dict(
+        required_input=[
+            {"name": "sensor_4_kw"},
+            {"name": "sensor_4_mw", "unit": "MW"},
+            {"name": "sensor_4_mwh", "unit": "MWh"},
+        ],
+        required_output=[
+            {"name": "sensor_4_kw"},
+            {"name": "sensor_4_mw"},
+            {"name": "sensor_4_mwh"},
+        ],
+        transformations=[],
+    )
+
+    reporter = PandasReporter(config=reporter_config)
+
+    start = datetime(2023, 1, 1, tzinfo=utc)
+    end = datetime(2023, 1, 2, tzinfo=utc)
+    input = [
+        dict(name="sensor_4_kw", sensor=s4),
+        dict(name="sensor_4_mw", sensor=s4),
+        dict(name="sensor_4_mwh", sensor=s4),
+    ]
+    output = [
+        dict(name="sensor_4_kw", sensor=s4),
+        dict(name="sensor_4_mw", sensor=s4),
+        dict(name="sensor_4_mwh", sensor=s4),
+    ]
+
+    report = reporter.compute(start=start, end=end, input=input, output=output)
+    result_kw = report[0]["data"]
+    result_mw = report[1]["data"]
+    result_mwh = report[2]["data"]
+
+    assert (
+        result_mw.event_value.values == result_kw.event_value.values / 1000
+    ).all()  # MW = kW / 1000
+    assert (
+        result_mwh.event_value.values == result_mw.event_value.values * 0.25
+    ).all()  # MWh = MW * 0.25 (resolution = 15 min)
```

Comment:
Does the Reporter actually convert anything, though?

Reply:
yes, it's converting the data from the sensors (`1 kW`) to:
- `MW` -> 0.001
- `MWh` -> 0.00025